Userland/Libraries/LibWeb/SVG/SVGScriptElement.cpp

```diff
@@ -19,4 +20,41 @@ void SVGScriptElement::initialize(JS::Realm& realm)
     set_prototype(&Bindings::ensure_web_prototype<Bindings::SVGScriptElementPrototype>(realm, "SVGScriptElement"));
 }
 
+void SVGScriptElement::visit_edges(Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_script);
+}
+
+// https://www.w3.org/TR/SVGMobile12/script.html#ScriptContentProcessing
+void SVGScriptElement::process_the_script_element()
+{
+    // 1. If the 'script' element's "already processed" flag is true or if the element is not in the
+    //    document tree, then no action is performed and these steps are ended.
+    if (m_already_processed || !in_a_document_tree())
+        return;
+
+    // FIXME: 2. If the 'script' element references external script content, then the external script content
+    //           using the current value of the 'xlink:href' attribute is fetched. Further processing of the
+    //           'script' element is dependent on the external script content, and will block here until the
+    //           resource has been fetched or is determined to be an invalid IRI reference.
+    if (has_attribute(HTML::AttributeNames::href)) {
+        dbgln("FIXME: Unsupported external fetch of SVGScriptElement!");
+        return;
+    }
+
+    // 3. The 'script' element's "already processed" flag is set to true.
+    m_already_processed = true;
+
+    // 4. If the script content is inline, or if it is external and was fetched successfully, then the
+    //    script is executed. Note that at this point, these steps may be re-entrant if the execution
+    //    of the script results in further 'script' elements being inserted into the document.
```

Comment:
The code doesn't seem to be checking these conditions before running the script?

Reply:
Left a FIXME for this. Long term, I'm considering whether I should try factor out common code between this and HTMLScriptElement into a common ScriptElement base class